templates/user/profile.tmpl

```diff
@@ -7,11 +7,11 @@
 					<div id="profile-avatar" class="content gt-df">
 					{{if eq .SignedUserID .ContextUser.ID}}
 						<a class="image" href="{{AppSubUrl}}/user/settings" data-tooltip-content="{{.locale.Tr "user.change_avatar"}}">
-							{{avatar $.Context .ContextUser 290}}
+							{{avatar $.Context .ContextUser 256}}
```

Comment:
I suggest to revert this, it doesn't take affect, it's just a no-op, and makes illusion.

Reply:
I thought I have left some comments, but maybe I forgot or the comment was lost.


```suggestion
							{{/* the size doesn't take affect, image side is actually caused by width:100%, which is decided by the parent containter */}}
							{{avatar $.Context .ContextUser 256}}
```

Reply:
The `100%` is just to override the fomantic style from `.ui.card img.avatar`, we can probably fix it via

````
width: unset;
height: unset;
```

to make the image's natural dimensions take effect. Long-term, I plan to remove the `card` module from Fomantic.

Reply:
I do not think "make the image's natural dimensions take effect" is right.

The avatar image element's side should be controlled by it's parent. The parent layout determine the children's size. But the children's width shouldn't stretch its parent.

Reply:
I applied that comment, feel free to edit.

Reply:
Hmm, I'm still a fan of not altering image dimensions in CSS because it's unnecessary because we are in full control via `width` and `height` attributes on the image. Let me try my method. I will add a comment to remove it later with the `.ui.card` removal.

Reply:
There could always be some cases that the image dimension needs to be altered.

For example: the image is too large or too small. There is no guarantee that the dimension is always exactly 256x256.

Reply:
Hmm ... maybe I get your point now. I guess you mean that "do not use width in CSS, but only set width on img tag" ?

Either looks good to me.

Reply:
Actually a global fix is needed because now that avatars can be non-square, we need to ensure they are contained in their dimensions, otherwise this happens as currently:

<img width="320" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/136005fc-b0f3-4dd1-8e13-9cc23c2dd26c">

Reply:
Yup, avatar images need `object-fit`

Reply:
Won't help. It is quite a though CSS nut to crack it seems. The only thing that works so far is explicit width/height via CSS. `height: attr(height px)` for some reason does nothing.

Reply:
It's weird, this works:

<img width="511" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/deaaeb60-fca9-4a63-bf93-c7a691faf334">

But this doesn't (i thought `unset` would be equivalen to absent):

<img width="513" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/ca08f57d-63c2-4c29-bc63-453d3f99a0a4">

Reply:
I think I may just remove the card module now, see no clean solution otherwise.

Reply:
Working on the removal.

Reply:
Done